Source/Common/FEXServerClient.cpp

```diff
@@ -100,41 +100,56 @@ namespace FEXServerClient {
     return GetServerLockFolder() + "RootFS.lock";
   }
 
-  std::string GetServerSocketFile() {
-    FEX_CONFIG_OPT(ServerSocketPath, SERVERSOCKETPATH);
-    if (ServerSocketPath().empty()) {
-      auto TmpDir = std::filesystem::temp_directory_path().string();
-
-      auto XDGRuntimeEnv = getenv("XDG_RUNTIME_DIR");
-      if (XDGRuntimeEnv) {
-        // If the XDG runtime directory works then use that instead.
-        TmpDir = XDGRuntimeEnv;
-      }
+  std::string GetServerTempFolder() {
+    // We need a server temporary folder that has the following requirements.
+    // - Can't be `/tmp/`
+    //   - systemd services use `PrivateTmp` feature to gives services their own tmp.
+    // - Can't be `$HOME/.fex-emu/`
+    //   - Can be mounted with a filesystem (sshfs) which can't handle mount points inside it.
+    auto Folder = std::filesystem::temp_directory_path().string();
+    auto XDGRuntimeEnv = getenv("XDG_RUNTIME_DIR");
+    if (XDGRuntimeEnv) {
+      // If the XDG runtime directory works then use that instead.
+      Folder = XDGRuntimeEnv;
+    }
 
-      return fmt::format("{}/{}.FEXServer.socket", TmpDir, ::geteuid());
+    if (FEXCore::Config::FindContainer() == "pressure-vessel") {
+      // In pressure-vessel the server location changes.
```

Comment:
Why? It's not apparent where this change comes from for someone without knowledge how pressure-vessel works. Is it an expectation of pressure-vessel? Or does it change the runtime environment somehow to make the other paths inaccessible?

Reply:
Cleared up the comment a bit here.
Yes, pressure-vessel if changing the runtime environment to make the paths inaccessible.
Pressure-vessel is a fancy chroot environment that does a whole bunch of directory rebinding which means we need to tip-toe around it a bit to be correct with our implementations.
`$XDG_RUNTIME_DIR` specifically disappears from our original view once it does a pivot_root.